tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -256,6 +260,30 @@ public:
                                                        std::vector<std::unique_ptr<RNTupleModel>> models = {});
 };
 
+// clang-format off
+/**
+\class ROOT::Experimental::RNTupleBaseProcessor
+\ingroup NTuple
+\brief Processor specializiation for processing a single RNTuple.
+*/
+// clang-format on
+class RNTupleBaseProcessor : public RNTupleProcessor {
```

Comment:
imho the keyword `Base` sounds more like `RNTupleProcessor` should derive from this class and not the opposite. Just as a thought, we might want to consider alternative namings, for example:

1. One way would be two switch the two class names, the base class would be called `RNTupleBaseProcessor` and the new implementation in this PR would be the `RNTupleProcessor`.
2. Or maybe we could think about some other keyword for this one, such as `Unit`, `Single`, `Lone`, `One`

Reply:
This is a good point. I am not particularly in favor of swapping the two names, because (what is currently) the `RNTupleProcessor` is the main interface users interact with so IMO this name should be preserved like that. I think `Unit` is a nice alternative, though. I will change it to this for now, and then we can come back and have another critical look at the naming once the interface has stabilized more.